chain-evm/src/transaction.rs

```diff
@@ -0,0 +1,49 @@
+use chain_core::{
+    packer::Codec,
+    property::{Deserialize, ReadError, Serialize, WriteError},
+};
+use ethereum::TransactionV2;
+use rlp::{decode, Decodable, DecoderError, Encodable, Rlp, RlpStream};
+use typed_bytes::ByteBuilder;
+
+/// Wrapper type for `ethereum::TransactionV2`, which includes the `EIP1559Transaction`, `EIP2930Transaction`, `LegacyTransaction` variants.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct EthereumTransaction(TransactionV2);
+
+impl EthereumTransaction {
+    /// Serialize the contract into a `ByteBuilder`.
+    pub fn serialize_in(&self, bb: ByteBuilder<Self>) -> ByteBuilder<Self> {
+        let bytes = self.rlp_bytes();
+        bb.u64(bytes.len() as u64).bytes(&bytes)
+    }
+}
+
+impl Decodable for EthereumTransaction {
+    fn decode(rlp: &Rlp<'_>) -> Result<Self, DecoderError> {
+        let tx = TransactionV2::decode(rlp)?;
+        Ok(EthereumTransaction(tx))
+    }
+}
+
+impl Encodable for EthereumTransaction {
+    fn rlp_append(&self, s: &mut RlpStream) {
+        self.0.rlp_append(s);
+    }
+}
+
+impl Serialize for EthereumTransaction {
```

Comment:
Actually I am not sure that these trait implementation as `Serialize`, `Deserialize` will also be needed for us, maybe we can just remove it at this point ?

Reply:
I think that we need these traits for Fragment-related ser/de, but I'm not sure.